long-ride-alerts/src/main/java/org/apache/flink/training/exercises/longrides/LongRidesExercise.java

```diff
@@ -98,17 +102,62 @@ public static void main(String[] args) throws Exception {
     @VisibleForTesting
     public static class AlertFunction extends KeyedProcessFunction<Long, TaxiRide, Long> {
 
+        private static final long RIDE_DURATION_ALERT_TIME_IN_MS = 1000 * 60 * 60 * 2;
+        private ValueState<TaxiRide> startTaxiRide;
+        private ValueState<TaxiRide> endTaxiRide;
```

Comment:
Make sure to make your state fields `transient`

Reply:
done